BrainPortal/app/controllers/userfiles_controller.rb

```diff
@@ -1150,18 +1150,31 @@ def download #:nodoc:
       specified_filename = "cbrain_files_#{current_user.login}.#{timestamp}"
     end
 
-    tot_size = 0
-
     # Find list of files accessible to the user
     userfiles_list = Userfile.find_accessible_by_user(filelist, current_user, :access_requested => :read)
-    tot_size = userfiles_list.inject(0) { |total, u| total + (u.size || 0) }
+
+    # When a new FileCollection is registered, the size and num_files attributes are temporarily set to nil.
+    # It takes few moments to calculate its size, so meanwhile we disallow downloading files of unknown,
+    # and potentially prohibitively large size
+    unsized = userfiles_list.detect { |u| u.size.nil? }
+    if unsized
+       flash[:error] = "Size of the file #{unsized.name} is not yet determined." +
+                       " Please try again latter.\n"
+       respond_to do |format|
+         format.html { redirect_to :action => :index }
+         format.json { render :json => { :error => flash[:error] } }
+      end
```

Comment:
I think the `end` keyword is not indented properly here? (Could be just github) Check please.

Reply:
I think the inside of the `if unsized` block is not indented correctly, on second thought.